htdocs/compta/facture/class/facture-rec.class.php

```diff
@@ -1286,6 +1286,10 @@ public function createRecurringInvoices($restrictioninvoiceid = 0, $forcevalidat
 			$sql .= ' AND rowid = '.((int) $restrictioninvoiceid);
 		}
 		$sql .= $this->db->order('entity', 'ASC');
+
+		//init hooks like in card to have the same result as if we have created the invoices manually
+		$hookmanager->initHooks(array('invoicecard','globalcard'));
```

Comment:
The context of a hook (param of initHooks) should not be set inside a CRUD method but by the caller (that know what is the context).
This means the beforeCreationOfRecurringInvoices can be created in different context.
If you need to have your hook working the same way for each context call, you must make your method working for each context set by caller.

Reply:
You are right @eldy  but I give you an example : when recurring invoices are created by cron job, the generated documents are not the same as if we have created it manually because of modules (like subtotal, autoliquidationtva, sitfac, ...) hooking on invoicecard context.
What would you do in those cases ? All modules have to hook on an other context ? Is there a better place where we could init hook 'invoicecard' context ?
Thanks

Reply:
So the context "cli" is a new context that module developers need to know and all modules concerning invoices must be modified to use this context.
The problem is that as long as the modules are not corrected, the documents will not output the same result by the card as by the cron (like subtotal or autoliquidationtva)
I'll let my modifications on my clients installation until then.
Thanks